packages/contracts/contracts/L1/ValidatorManager.sol

```diff
@@ -0,0 +1,637 @@
+            uint256 l2Timestamp = L2_ORACLE.nextOutputMinL2Timestamp();
+            if (block.timestamp >= l2Timestamp) {
+                uint256 elapsed = block.timestamp - l2Timestamp;
+                // If the current time exceeds one round time, it is a public round.
+                if (elapsed > ROUND_DURATION_SECONDS) {
+                    return Constants.VALIDATOR_PUBLIC_ROUND_ADDRESS;
+                }
+            }
+
+            return _nextPriorityValidator;
+        } else {
+            return TRUSTED_VALIDATOR;
+        }
+    }
+
+    /**
+     * @inheritdoc IValidatorManager
+     */
+    function getStatus(address validator) public view returns (ValidatorStatus) {
+        if (!_validatorInfo[validator].isInitiated) {
+            return ValidatorStatus.NONE;
+        }
+
+        if (_jail[validator] != 0) {
+            return ValidatorStatus.IN_JAIL;
+        }
+
+        if (ASSET_MANAGER.totalValidatorKro(validator) < MIN_REGISTER_AMOUNT) {
+            return ValidatorStatus.INACTIVE;
+        }
+
+        bool started = _validatorTree.nodeMap[validator] > 0;
+
+        // To prevent all MIN_START_AMOUNT is fulfilled with KRO in KGH which is not subject to slash,
+        // enable to start the validator when real asset satisfies the threshold.
+        if (
+            ASSET_MANAGER.reflectiveWeight(validator) - ASSET_MANAGER.totalKroInKgh(validator) <
+            MIN_START_AMOUNT
+        ) {
+            if (!started) {
+                return ValidatorStatus.ACTIVE;
+            }
+            return ValidatorStatus.STARTED;
+        }
+
+        if (!started) {
+            return ValidatorStatus.CAN_START;
+        }
+        return ValidatorStatus.CAN_SUBMIT_OUTPUT;
+    }
+
+    /**
+     * @notice Returns the no submission count of given validator.
+     *
+     * @param validator Address of the validator.
+     *
+     * @return The no submission count of given validator.
+     */
+    function noSubmissionCount(address validator) public view returns (uint8) {
+        return _validatorInfo[validator].noSubmissionCount;
+    }
+
+    /**
+     * @notice Returns the total weight of started validators.
+     *
+     * @return The total weight of started validators.
+     */
+    function startedValidatorTotalWeight() public view returns (uint120) {
+        return _validatorTree.nodes[_validatorTree.root].weightSum;
+    }
+
+    /**
+     * @notice Internal function to add output submission rewards to the vaults of finalized output
+     *         submitters.
+     *
+     * @return Whether the reward distribution is done at least once or not.
+     */
+    function _distributeReward() internal returns (bool) {
+        uint256 outputIndex = L2_ORACLE.latestFinalizedOutputIndex() + 1;
+        uint256 latestOutputIndex = L2_ORACLE.latestOutputIndex();
+
+        if (!L2_ORACLE.VALIDATOR_POOL().isTerminated(outputIndex)) {
+            return false;
+        }
```

Comment:
Is it required for overlapping periods?

Reply:
Yes, please see [here](https://github.com/kroma-network/kroma/pull/277#discussion_r1537520598).